core/embed/trezorhal/stm32f4/backlight_pwm.h

```diff
@@ -1,17 +1,60 @@
-
-#ifndef CORE_BACKLIGHT_H
-#define CORE_BACKLIGHT_H
-
-#include "common.h"
-
-int backlight_pwm_set(int val);
-
-int backlight_pwm_get(void);
-
-void backlight_pwm_init(void);
-
-void backlight_pwm_reinit(void);
-
-void backlight_pwm_set_slow(void);
-
-#endif  // CORE_BACKLIGHT_H
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef TREZORHAL_BACKLIGHT_H
+#define TREZORHAL_BACKLIGHT_H
+
+// Level of backlight in range 0-255
+typedef int backlight_level_t;
```

Comment:
The typedef seems pointless. Though i would be for changing the backlight level value type to uint8_t

Reply:
I reverted this change because I didn't find a strong enough argument for it. Until we maintain the old drivers and have a dual-purpose `display_backlight()` function (acting as both a setter and getter by using -1 to get the value), It's better to continue to use the `int` type. Once we remove the legacy driver, I will refactor it, probably switch to `uint8_t`, and maybe reintroduce `backlight_level_t`;-). see 6f57cc88f20e8cf3c56b60557e55df37940f56ca